crates/gossip/src/state.rs

```diff
@@ -0,0 +1,194 @@
+use crate::protocol::{
+    K2GossipAcceptMessage, K2GossipAgentsMessage, K2GossipNoDiffMessage,
+};
+use kitsune2_api::{AgentId, K2Error, K2Result, Url};
+use rand::RngCore;
+
+/// The state of a gossip round.
+#[derive(Debug)]
+pub(crate) struct GossipRoundState {
+    /// The agent id of the other party who is participating in this round.
+    pub session_with_peer: Url,
+
+    /// The time at which this round was initiated.
+    ///
+    /// This is used to apply a timeout to the round.
+    #[allow(dead_code)]
+    started_at: std::time::Instant,
+
+    /// The session id of this round.
+    ///
+    /// Must be randomly chosen and unique for each initiated round.
+    pub session_id: bytes::Bytes,
```

Comment:
Here to prevent message replay etc. Helps us avoid getting into unexpected states by receiving messages that were created before a timeout or anything like that